src/DynamoCore/Search/NodeSearchModel.cs

```diff
@@ -13,6 +12,33 @@ namespace Dynamo.Search
     /// </summary>
     public class NodeSearchModel : SearchLibrary<NodeSearchElement, NodeModel>
     {
+        private PreferenceSettings preferences;
+
+        private bool isDetailedMode;
+
+        /// <summary>
+        ///  The property specifies which layout(detailed or compact) is used.
+        /// </summary>
+        public bool IsDetailedMode
+        {
+            get { return isDetailedMode; }
+            set
+            {
+
+                isDetailedMode = value;
+                preferences.ShowDetailedLayout = isDetailedMode;
+            }
```

Comment:
You can set the value of preferences.ShowDetailedLayout here. Get rid of private variable. Also,
 from the constructor, I see prefrences can be null. That case you can check for null and return true / false by default.